pkg/domain/runaway.go

```diff
@@ -43,7 +45,11 @@ func (do *Domain) initResourceGroupsController(ctx context.Context, pdClient pd.
 		return nil
 	}
 
-	control, err := rmclient.NewResourceGroupController(ctx, uniqueID, pdClient, nil, rmclient.WithMaxWaitDuration(runaway.MaxWaitDuration))
+	keyspaceID := constants.NullKeyspaceID
+	if v := keyspace.GetKeyspaceIDBySettings(); v != nil {
```

Comment:
Should not rely on this next-gen only interface, better use `store.codec.KeyspaceID()` instead